### PR TITLE
Java out of process requirements

### PR DESCRIPTION
**Pull request by ancaantochi**

### CODE REVIEW

File: proxy/gateway/java/devdoc/java_out_of_process_hld.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+Java Out of Process Module API
+==============================
+
+Overview
+--------
+
+This document describes the API that enables communication between an out of process Java module and Azure IoT Gateway. 
+An out of process module should implement `IGatewayModule` interface or extend the `GatewayModule` abstract class as it does for an in-process module, more details in [Java binding high level design](java_binding_hld.md). In addition, the API provides `ProxyGateway` which is a proxy for the IGatewayModule implementation and handles the communication between the Gateway and the out of process module.
```
Comment (darobs) [inline proxy/gateway/java/devdoc/java_out_of_process_hld.md]: >s it does f [](start = 115, length = 11)

I like it when the markdown files have line breaks around 80 columns, although I'm not always consistent about it.  It does make it easier to read in some applications. (like code flow)

File: proxy/gateway/java/devdoc/java_out_of_process_hld.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+-  *data* messages which are sent between modules via the Gateway.
+
+
+![](media/java_oop_module_hld.png)
+
+Receive out of process messages from the Gateway
+------------------------------------------------
+
+In order to make a module to run out of process, create an instance of `ProxyGateway`, pass the configuration that contains the unique identifier to attach to the Gateway and then call `attach`. 
+Also the Gateway configuration has to be changed for that specific module to set the loader as an out of process loader and set the same unique identifier.
```
Comment (darobs) [inline proxy/gateway/java/devdoc/java_out_of_process_hld.md]: >configuration has [](start = 17, length = 17)

Include an example?

File: proxy/gateway/java/devdoc/java_out_of_process_hld.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+``` json
+"modules" : [
+{ 
+    "name" : "outprocess_module",
+    "loader" : {
+        "name" : "outprocess"
+        "entrypoint" : {
+            "activation.type" : "none",
+            "control.identifier" : "outprocess_module_control_unique_id",
```
Comment (darobs) [inline proxy/gateway/java/devdoc/java_out_of_process_hld.md]: >identifier [](start = 21, length = 10)

Is it just "id"?

File: proxy/gateway/java/devdoc/module_configuration_requirements.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+**SRS_MODULE_CONFIGURATION_24_003: [** It shall return the control channel identification. **]**
+
+## getModuleClass
+```java
+public Class<? extends IGatewayModule> getModuleClass()
+```
+**SRS_MODULE_CONFIGURATION_24_004: [** It shall return the module class. **]**
+
+## getModuleClass
```
Comment (darobs) [inline proxy/gateway/java/devdoc/module_configuration_requirements.md]: >getModuleClass [](start = 3, length = 14)

getVersion?

**Comment by darobs** [comment]
:shipit:

File: proxy/gateway/java/devdoc/java_out_of_process_hld.md (hunk 3 above)
Comment (ancaantochi) [inline proxy/gateway/java/devdoc/java_out_of_process_hld.md]: changed, thanks

File: proxy/gateway/java/devdoc/module_configuration_requirements.md (hunk 4 above)
Comment (ancaantochi) [inline proxy/gateway/java/devdoc/module_configuration_requirements.md]: done

File: proxy/gateway/java/devdoc/java_out_of_process_hld.md (hunk 1 above)
Comment (ancaantochi) [inline proxy/gateway/java/devdoc/java_out_of_process_hld.md]: I think 80 columns is a bit too small.

**Pull request merged**